openmrs-19/src/main/resources/webapp/js/controllers.js

```diff
@@ -0,0 +1,156 @@
+(function () {
+    'use strict';
+
+    /* Controllers */
+
+    var controllers = angular.module('openmrs19.controllers',[]);
+
+    controllers.controller('OpenMRSSettingsCtrl', function($scope, OpenMRSConfig) {
+
+        $scope.configs = undefined;
+        $scope.selectedConfig = undefined;
+        $scope.configOutdated = false;
+        $scope.newPatientIdentifierType={};
+
+        $scope.getDefault = function () {
+            var i;
+            for (i = 0; i < $scope.configs.configs.length; i += 1) {
+                if ($scope.configs.configs[i].name === $scope.configs.defaultConfigName) {
+                    return $scope.configs.configs[i];
+                }
+            }
+            return undefined;
+        };
+
+         $scope.getConfigs = function () {
+             OpenMRSConfig.get(
+             function success(data) {
+                 $scope.configs = data;
+                 $scope.configs.configs.patientIdentifierTypeNames=[];
+                 $scope.configOutdated = false;
+                 if ($scope.selectedConfig === undefined) {
+                    $scope.selectedConfig = $scope.getDefault();
+                 }
+            });
+        };
+
+        $scope.getConfigs();
+
+        $scope.addConfig = function () {
+            $scope.selectedConfig = {
+                'name':'',
+                'openMrsUrl':'',
+                'username':'',
+                'password':'',
+                'motechPatientIdentifierTypeName':'',
+                'patientIdentifierTypeNames':[]
+            };
+        };
+
+        $scope.hasValue = function(prop) {
+            return $scope.selectedConfig[prop] !== null &&
```

Comment:
this is basically equivalent to if ($scope.selectedConfig[prop]) - we do not need the method